python/pfs/drp/stella/lsf.py

```diff
@@ -0,0 +1,688 @@
+            Array containing the LSF inserted.
+        """
+        xx = np.arange(self.length)
+        return self.norm*gaussian(xx - center, self.sigma)
+
+    def computeKernel(self, center):
+        """Return a kernel for the nominated position
+
+        Besides the difference in return types from the ``computeArray``
+        method (this returns a `Kernel1D`, while ``computeArray`` returns a
+        an array), this method provides a model for the LSF centered at zero.
+        This serves as the foundation for the other methods in the class.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF. Not that it matters, as it's
+            the same everywhere.
+
+        Returns
+        -------
+        kernel : `Kernel1D`
+            Kernel with a model of the LSF at the nominated position.
+        """
+        return self.kernel
+
+    def computeShape1D(self, center):
+        """Return the standard deviation at the nominated position
+
+        The standard deviation is a measure of the width of the LSF.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF. Not that it matters, as it's
+            the same everywhere.
+
+        Returns
+        -------
+        stdev : `float`
+            Standard deviation of the LSF at the nominated position.
+        """
+        return self.sigma
+
+    def __reduce__(self):
+        """Pickling"""
+        return self.__class__, (self.length, self.sigma)
+
+
+class FixedEmpiricalLsf(Lsf):
+    """A constant empirical LSF
+
+    Parameters
+    ----------
+    kernel : `Kernel1D`
+        The kernel to use.
+    length : `int`
+        Array length.
+    interpolator : callable, optional
+        Interpolator factory for kernel. Calling this with arrays of ``x`` and
+        ``y`` should produce an interpolator that can be called with an array of
+        ``x`` values at which to interpolate.
+    """
+    def __init__(self, kernel, length, interpolator=DEFAULT_INTERPOLATOR):
+        super().__init__(length, interpolator=interpolator)
+        self.kernel = kernel
+
+    def computeKernel(self, center):
+        """Return a kernel for the nominated position
+
+        Besides the difference in return types from the ``computeArray``
+        method (this returns a `Kernel1D`, while ``computeArray`` returns a
+        an array), this method provides a model for the LSF centered at zero.
+        This serves as the foundation for the other methods in the class.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF. Not that it matters, as it's
+            the same everywhere.
+
+        Returns
+        -------
+        kernel : `Kernel1D`
+            Kernel with a model of the LSF at the nominated position.
+        """
+        return self.kernel
+
```

Comment:
Why don't you overload `computeApertureFlux`?

Reply:
I don't remember if it's because I'm lazy or if I wanted to sum the pixels. In any case, I don't see it being used very much, and so I'm going to default to being lazy.